lib/debugger/src/dbg_ieval.erl

```diff
@@ -1807,6 +1985,31 @@ merge_bindings([{Name,V}|B1s], B2s, Ieval) ->
 merge_bindings([], B2s, _Ieval) ->
     B2s.
 
+zip_add_bindings(Bs1, Bs2) when is_map(Bs1), is_map(Bs2) ->
+    zip_add_bindings_map(maps:keys(Bs2), Bs1, Bs2);
+zip_add_bindings(Bs1, Bs2) when is_list(Bs1), is_list(Bs2) ->
+    zip_add_bindings1(orddict:to_list(Bs1), Bs2).
+
+zip_add_bindings_map([Key | Keys], Bs1, Bs2) ->
+    case {Bs1, Bs2} of
+        {#{Key := Same}, #{Key := Same}} -> zip_add_bindings_map(Keys, Bs1, Bs2);
+        {#{Key := _}, _} -> nomatch;
+        {_, #{Key := Value}} -> zip_add_bindings_map(Keys, Bs1, Bs2#{Key => Value})
```

Comment:
This looks redundant, as does the final `maps:merge/2`.

Reply:
Yep that's unnecessary. I mixed up Bs1 and Bs2 and got a bit confused about why the merge is needed.